python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -652,3 +653,21 @@ def _isin_datetimelike(
 
     res = lhs._obtain_isin_result(rhs)
     return res
+
+
+def to_iso_calendar(self):
+    formats = ["%G", "%V", "%u"]
+    if isinstance(self, (cudf.Index, pd.Index)):
+        iso_params = [
+            self._values.as_string_column(self._values.dtype, fmt)
+            for fmt in formats
+        ]
+        index = self._values
+    elif isinstance(self.series, (cudf.Series, pd.Series)):
+        iso_params = [self.strftime(fmt) for fmt in formats]
+        index = self.series.index
+
+    data = dict(zip(["year", "week", "day"], iso_params))
+    isoSeries = cudf.DataFrame(data, index=index, dtype=np.int32)
+
+    return isoSeries
```

Comment:
```suggestion
    return cudf.DataFrame(data, index=index, dtype=np.int32)
```

`isoSeries` will mislead someone not familiar to this code to think that it is a series, but it is a dataframe here.